### PR TITLE
Adds EIP712 Signature support to the platform

### PR DESCRIPTION
**Pull request by akeem**
This pull request adjusts functionality in the application for the better. This description is worth reading as it outlines the changes made, why, and provides resources to better understand the decisions made. 

We've moved the address owner verification from using a JWT a ECDA signature to the signature of Typedata based on the standard as outlined via EIP 712.

Over all this provides a much better user experience as people (Lock purchasers/creators) have a clearer understanding of the information requesting to be signed. This is a huge improvement over using the standard signature methods, as people are asked to sign opaque data. 

Here are some resources outlining the specification and general information that helped to steer the direction:

* https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md
* https://medium.com/metamask/eip712-is-coming-what-to-expect-and-how-to-use-it-bb92fd1a7a26
* https://medium.com/metamask/scaling-web3-with-signtypeddata-91d6efc8b290

screen shot 2019-02-11 at 2 51 16 pm

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)

### CODE REVIEW

**Comment by akeem** [comment]
A pull request with the dependencies is being created as a precursor to this adjustment

File: locksmith/__tests__/signature_validation_middleware.test.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
+var httpMocks = require('node-mocks-http')
```
Comment (cellog) [inline locksmith/__tests__/signature_validation_middleware.test.js]: move to camelCase for filename?

File: locksmith/src/signatureValidationMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+var sigUtil = require('eth-sig-util')
+const ethJsUtil = require('ethereumjs-util')
+const Base64 = require('./utils/base64')
+
+const extractToken = req => {
+  if (
```
Comment (cellog) [inline locksmith/src/signatureValidationMiddleware.js]: maybe add a 
```js
  const auth = req.headers.authorization && req.headers.authorization.split(' ')
```
here to DRY it up

File: unlock-app/src/middlewares/walletMiddleware.js [hunk 3]
```diff
@@ -125,15 +125,29 @@ export default function walletMiddleware({ getState, dispatch }) {
         ensureReadyBefore(() => {
           walletService.createLock(action.lock, getState().account.address)
           if (config.services.storage) {
-            generateJWTToken(walletService, getState().account.address, {
-              lock: action.lock,
-            })
+            generateSignature(
+              walletService.web3,
+              getState().account.address,
```
Comment (cellog) [inline unlock-app/src/middlewares/walletMiddleware.js]: Didn't catch this before, but `account` can be `null`. You'll want to bail out at the top of `ensureReadyBefore` with false if `getState().account` is falsy

Comment (akeem) [reply unlock-app/src/middlewares/walletMiddleware.js]: @cellog Do you now if the `walletService.ready` check inside of the declaration of `ensureReadyBefore` would handle the general case of an account not being available

Comment (cellog) [reply unlock-app/src/middlewares/walletMiddleware.js]: probably. I'll leave this as resolved because of that.

**Pull request merged**